### ISSUE TITLE
Conda environment creation fails when executed from the release distribution

### ISSUE THREAD

**Issue by byoo** [broadinstitute/gatk#4209]
First of all, thank you so much for your hard work! I am excited to explore GATK 4.0.0.0.
I would like to report minor issues in setting up optional dependencies for GATK4.

First, `gatkcondaenv.yml` defines `gatkPythonPackageArchive.zip` location as
`build/gatkPythonPackageArchive.zip`
The path doesn't seem to work seamlessly with the release archive, though the path makes sense in the root directory of source repo. Unzipping the package file locates both files in the same directory and, to me, it is natural to create the conda environment in the directory.

Second, it would be more convenient to include `install_R_packages.R` in the release like gatkcondaenv.yml for python dependencies. Also, I had an issue running `install_R_packages.R` for fresh compiled R-3.2.5. The way `install_R_packages.R` install `optparse` prevents its dependency (the getopt package) from being automatically installed, because (I guess) repos argument is set to NULL. I think the helper script needs to install the getopt package as well.
```
optparseUrl="http://cran.r-project.org/src/contrib/Archive/optparse/optparse_1.3.2.tar.gz"
if (!("optparse" %in% rownames(installed.packages()))) {
  install.packages(optparseUrl, repos=NULL, type="source")
}
```

Finally, it would be convenient to depend on a R version available in conda, so that R dependency can be a part of the gatk conda environment. I wasn't able to get the recommended R-3.2.5 from conda.

**Comment by droazen** [comment]
Thanks for raising these issues @byoo -- @cmnbroad would you be able to address them?

**Comment by cmnbroad** [comment]
@byoo Thanks for reporting these issues. I'm going to move the R issue into a separate ticket, since its a bigger change than the python fix.

I'm not sure I understand this part, though:

> Unzipping the package file locates both files in the same directory and, to me, it is natural to create the conda environment in the directory.

There should be no need to manually unzip the package archive (maybe you did this as part of the workaround for bad path in the .yml ?). And you should be able to direct conda to create the env wherever you want it - we just suggested the simplest possible default case. Let me know if I'm misunderstanding your suggestion.

**Comment by byoo** [comment]
@cmnbroad Sorry I should have been clearer. I meant unzipping the gatk release file. Unzipping the gatk release file places both files in the sample directory as below. I tried to create the gatk conda environment in the directory, which was failed due to the path issue.
```
$ unzip src/gatk-4.0.0.0.zip
Archive:  src/gatk-4.0.0.0.zip
   creating: gatk-4.0.0.0/
  inflating: gatk-4.0.0.0/gatk-package-4.0.0.0-local.jar
  inflating: gatk-4.0.0.0/gatk-package-4.0.0.0-spark.jar
  inflating: gatk-4.0.0.0/gatk
  inflating: gatk-4.0.0.0/README.md
  inflating: gatk-4.0.0.0/gatk-completion.sh
  inflating: gatk-4.0.0.0/gatkcondaenv.yml
  inflating: gatk-4.0.0.0/gatkPythonPackageArchive.zip
$ cd gatk-4.0.0.0
$ conda env create -n gatk -f gatkcondaenv.yml
```

It appears simpler to have conda manage both Python and R dependencies. I am not sure if it's easily possible to move away from R-3.2.5 though. Thank you!

**Comment by lbergelson** [comment]
I think we could probably change away from 3.25, I have 3.4.3 installed on my machine and tests run fine.  We just chose one arbitrarily that worked and set it at that so that it wouldn't change out from under us.

**Comment by cmnbroad** [comment]
@lbergelson Do we actually have any tests that depend on R besides the RScriptExecutor tests (which don't do much). Also wondering if there would be inconsistencies with Picard R requirements.

**Comment by cmnbroad** [comment]
@byoo Thanks for the clarification - makes sense.

**Comment by lbergelson** [comment]
@cmnbroad There used to be, but I don't actually know for sure.  I'm not sure how to figure it out either without just deleting R from the installation and running the test suite...  I didn't know picard required R?

**Comment by samuelklee** [comment]
There are tests for the CNV plotting tools, which use RScriptExecutor and uniquely require some of the R dependencies.  There's also a test for the HMM code, which uses an HMM R package to generate expected results.

I'd assume that tests for the other tools that use RScriptExecutor would also fail, but I don't know if they rely on any special dependencies or any particular version of R.

**Comment by samuelklee** [comment]
OK, I think I accidentally removed the `getopt` dependency in #3935.  Not sure why tests didn't fail as expected.

**Comment by samuelklee** [comment]
Hmm, I also realize that I did not update the base Docker image at that time---although I commented in the PR that we should...and I don't think we updated it before release.  So I *think* the latest base and the current release Docker contain the getopt dependency, although it was removed from the installation script.

**Comment by samuelklee** [comment]
Going to build and push a new 1.2.2 base image off of https://github.com/broadinstitute/gatk/tree/sl_getopt, which restores the dependency as it was previously.

**Comment by samuelklee** [comment]
Regarding the non-Docker integration tests failing earlier today, I think this was because the R packages were added to the Travis cache in #3101.  @cmnbroad cleared the cache to see if we could reproduce a compiler error introduced in #3934 on Travis (for the record, we could reproduce it on my local Ubuntu machine and gsa5, but not on Travis).  This removed the cached getopt dependency, which then caused tests to fail.  See #4246.

**Comment by samuelklee** [comment]
Closed in #4233 and #4246, but see #4250.
